enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/server/CoreServerModule.java

```diff
@@ -177,10 +178,13 @@ private OnlineBackupKernelExtension pickBackupExtension( NeoStoreDataSource data
 
         CoreStateDownloader downloader = new CoreStateDownloader( localDatabase, servicesToStopOnStoreCopy,
                 remoteStore, catchUpClient, logProvider, storeCopyProcess, coreStateMachinesModule.coreStateMachines,
-                snapshotService, commandApplicationProcess, topologyService );
+                snapshotService, topologyService );
+
+        CoreStateDownloaderService downloadService = new CoreStateDownloaderService( platformModule
```

Comment:
Make this a LifeCycle and add it to platformModule.life at some appropriate place. Downloads should not be allowed when stopped and stopping could interrupt an on-going download.